avocado/core/loader.py

```diff
@@ -1027,6 +1014,68 @@ def get_decorator_mapping():
         return {test.ExternalRunnerTest: output.TERM_SUPPORT.healthy_str}
 
 
+class PythonUnittestLoader(FileLoader):
+    name = "py_unittest"
+
+    __not_test_str = ("Does not look like a python unittest")
+
+    @staticmethod
+    def get_type_label_mapping():
+        return {NotATest: 'NOT_A_TEST',
+                MissingTest: 'MISSING',
+                BrokenSymlink: 'BROKEN_SYMLINK',
+                AccessDeniedPath: 'ACCESS_DENIED',
+                test.Test: 'INSTRUMENTED',
+                test.PythonUnittest: 'PyUNITTEST'}
+
+    @staticmethod
+    def get_decorator_mapping():
+        return {NotATest: output.TERM_SUPPORT.warn_header_str,
+                MissingTest: output.TERM_SUPPORT.fail_header_str,
+                BrokenSymlink: output.TERM_SUPPORT.fail_header_str,
+                AccessDeniedPath: output.TERM_SUPPORT.fail_header_str,
+                test.PythonUnittest: output.TERM_SUPPORT.healthy_str}
+
+    def _make_tests(self, test_path, list_non_tests, subtests_filter=None):
+        def ignore_broken(klass, uid, description=None):
+            """ Always return empty list """
+            return []
+
+        if list_non_tests:  # return broken test with params
+            make_broken = self._make_test
+        else:  # return empty set instead
+            make_broken = ignore_broken
+        try:
+            # Python unittests
+            assert (test_path.endswith(".py") or test_path.endswith(".pyc") or
+                    test_path.endswith(".pyo"))
+            old_dir = os.path.abspath(os.path.curdir)
+            try:
+                _test_dir = os.path.abspath(os.path.dirname(test_path))
+                _test_name = os.path.basename(test_path)
+                os.chdir(_test_dir)
+                python_unittests = self._find_python_unittests(_test_name)
+            finally:
+                os.chdir(old_dir)
+            if python_unittests:
+                return [(test.PythonUnittest, {"name": name,
+                                               "test_dir": _test_dir})
+                        for name in python_unittests
+                        if not (subtests_filter and
+                                not subtests_filter.search(name))]
+            else:
+                return make_broken(NotATest, test_path,
+                                   self.__not_test_str)
+
+        # Since a lot of things can happen here, the broad exception is
+        # justified. The user will get it unadulterated anyway, and avocado
+        # will not crash.
+        except BaseException as details:  # Ugly python files can raise any exc
+            if isinstance(details, KeyboardInterrupt):
+                raise  # Don't ignore ctrl+c
+            return make_broken(NotATest, test_path, self.__not_test_str)
+
+
```

Comment:
Oh, that was fast!
Yes, that's precisely what I was expecting. Please squash it.